packages/nocms-plugin-image-resources/test/createImageResourceProvider.spec.js

```diff
@@ -0,0 +1,94 @@
+import assert from 'assert';
+// import { createImageResourceProvider } from '../lib/createImageResourceProvider.js';
+import { activate } from '../lib/index.js';
+import fs from 'fs';
+
+const filePaths = [
+	'test/test-images/50mm-249.jpg',
+	'test/test-images/8-tallet.jpg'
+];
+
+let resourceProviders = [];
+let pluginActivationContext;
+const fakeFiles = [];
+const writtenFiles = [];
+
+function fileToBuffer(path){
+	return new Promise((resolve, reject) => {
+		fs.readFile(path, (err, data) => {
+			if( err ) {
+				return reject(err);
+			}
+			return resolve(data);
+		});
+	});
+}
+
+function readFileFake(fakePath) {
+	return new Promise((resolve, reject) => {
+		fakeFiles.forEach( fakeFile => {
+			if(fakeFile.path === fakePath){
+				return resolve(fakeFile.data);
+			}
+		});
+		return reject('wat');
+	});
+}
+
+function writeFileFake(filePath, data, options){
+	writtenFiles[filePath] = data;
+	return Promise.resolve();
+}
+
+function findFilesFake(globPattern) {
+	return new Promise(resolve => {
+		resolve(fakeFiles.map(fakeFile => fakeFile.path));
+	});
+}
+
+describe('nocms-plugin-image-resources', () => {
+	before(async () => {
```

Comment:
Here I think I would:
1.  create a scoped `const fakeFiles = [];` (on line 50, after describe - the one that you have in global scope).
2. push like you do now.

Reply:
I need the `fakeFiles` in the global scope to expose them to the faked read function at the moment.